docs: update + centralize project docs (#2598)
adobe/spectrum-css: README.md

<img src="assets/spectrum-css_illustration_desktop.png">

> ## Quick start

Reviewer: ```suggestion
## Quick start guide

This example will show you how to quickly install Spectrum CSS and use it to display a component.
```
Thoughts on adjusting the name to "Quick start guide"?

I think it needs a small intro here to explain what this guide does; the text I've suggested could probably use some work and elaboration.

> ## Quick start

Author: Oh yes, I like that idea. I'll move it into it's own file!

> ### Node/yarn

Author: I moved this section over from `site/get-started.pug` and did some minor rewording.

Install the components you want along with their dependencies. Here's an example using the button component:

```shell
yarn add -DW @spectrum-css/tokens @spectrum-css/typography @spectrum-css/page @spectrum-css/icon @spectrum-css/button
```

Spectrum CSS components utilize custom properties in order to change themes and scales. For these to apply, a couple of classes need to be added to the document’s `<html>` tag based on the [visual language](https://github.com/adobe/spectrum-css?tab=readme-ov-file#visual-language), [scale](https://github.com/adobe/spectrum-css?tab=readme-ov-file#scales), and [theme](https://github.com/adobe/spectrum-css?tab=readme-ov-file#themes-colorstops) you wish to use. For example, the following code snippet will display styling for the default Spectrum visual language using medium scale and light color theme:

```html
<html class="spectrum spectrum--medium spectrum--light">
```

> Use the `index.css` files in your project to include component styles and all of the global context variable definitions (Spectrum and Express contexts, background theme/colorstop, platform scaling, etc.) for the component (if you don't need all of the global context definitions, peek at the docs for [including assets](https://github.com/adobe/spectrum-css?tab=readme-ov-file#including-assets)). Use this file by including the stylesheet (plus the stylesheets for the dependencies) in the `<head>` tag:

Reviewer: ```suggestion
Use the `index.css` files in your project to include component styles and all of the global context variable definitions (Spectrum and Express contexts, background theme/colorstop, platform scaling, etc.) for the component. If you don't need all of the global context definitions, peek at the docs for [including assets](https://github.com/adobe/spectrum-css?tab=readme-ov-file#including-assets)). Use this file by including the stylesheet and the stylesheets for the dependencies in the `<head>` tag:
```
A few edits to remove some parenthesis.

I'm thinking that the sentence "_global context variable definitions (Spectrum and Express contexts, background theme/colorstop, platform scaling, etc.)_" might be a little wordy and may need some adjustment to the terminology? Should we say something about global custom properties from the "Spectrum CSS tokens" package? Or reference "[design tokens](https://spectrum.adobe.com/page/design-tokens/)" for the design system?

Author: Good point, this is wordy. I'm trying not to repeat too much here in the quick start guide that folks can find in the README. I tried to simplify this a bit and link to relevant parts of the README. Let me know if the change addresses your concerns!

```html
<head>
  <!-- Include global tokens first -->
  <link rel="stylesheet" href="node_modules/@spectrum-css/tokens/dist/index.css"/>

  <!-- Include index.css for all components you're using -->
  <link rel="stylesheet" href="node_modules/@spectrum-css/page/dist/index.css"/>
  <link rel="stylesheet" href="node_modules/@spectrum-css/button/dist/index.css"/>
</head>
```

> Inside the <body> tag, add the markup for your component (Spectrum button in our example). The example also includes the CSS class names spectrum-Button--fill spectrum-Button--accent, to use the accent variant:

Reviewer: ```suggestion
Inside the `<body>` tag, add the markup for your component (Spectrum button in our example). The example also includes the CSS class names `spectrum-Button--fill` and `spectrum-Button--accent`, to use the accent variant:
```
The body tag was being rendered here.

```html
<button class="spectrum-Button spectrum-Button--fill spectrum-Button--accent spectrum-Button--sizeM">
  <span class="spectrum-Button-label">Button</span>
</button>
```

To put it all together, your final html file will look like this:

```html
<html class="spectrum spectrum--light spectrum--medium">
  <head>
    <link rel="stylesheet" href="node_modules/@spectrum-css/tokens/dist/index.css"/>
    <link rel="stylesheet" href="node_modules/@spectrum-css/page/dist/index-vars.css">
    <link rel="stylesheet" href="node_modules/@spectrum-css/button/dist/index-vars.css">
  </head>
  <body>
    <button class="spectrum-Button spectrum-Button--fill spectrum-Button--accent spectrum-Button--sizeM">
      <span class="spectrum-Button-label">Button</span>
    </button>
  </body>
</html>
```

> ### CDN

Reviewer: ```suggestion
### Option 2: Include files from a CDN
```

Another way to include Spectrum CSS components in your project is to use a CDN, [unpkg](https://unpkg.com/) for example, to link to the components you want, plus their dependencies, in the `<head>` tag of your HTML. Here's an example using the button component:

```html
<head>
  <!-- Include global tokens first -->

>   <link rel="stylesheet" href="https://unpkg.com/browse/@spectrum-css/tokens/"/>

Reviewer 2: ```suggestion
  <link rel="stylesheet" href="https://unpkg.com/@spectrum-css/tokens/dist/index.css" />
```

  <!-- Include index.css for all components you're using -->

>   <link rel="stylesheet" href="https://unpkg.com/browse/@spectrum-css/page/"/>
>   <link rel="stylesheet" href="https://unpkg.com/browse/@spectrum-css/button/"/>

Reviewer 2: ```suggestion
  <link rel="stylesheet" href="https://unpkg.com/@spectrum-css/page/dist/index.css"/ >
  <link rel="stylesheet" href="https://unpkg.com/@spectrum-css/button/dist/index.css" />
```

</head>

> ```

Author: One of our [users requested](https://adobedesign.slack.com/archives/C5N154FEY/p1710946376240839?thread_ts=1710340701.269509&cid=C5N154FEY) we show how to use a CDN to include Spectrum CSS in a project. Is this how we would do that?

Reviewer 2: I suggested changes for the URLs, but yes, this is mostly correct. We need to point _directly_ to a specific CSS asset in the `dist`, and here I'm suggesting that it's the `index.css` that folks would want to use.

Also, unpkg should just resolve the request to the latest version that's been published to npm, so no need to include the actual version number (unless someone wanted to use a specific version number)

* 📖 **Robust documentation**: Spectrum CSS is designed to be used in partnership with [Spectrum's detailed usage guidelines](https://spectrum.adobe.com/).
* 🎨 **Flexible**: Our CSS is customizable, powerful, and designed to work with any JavaScript framework.
* 🧪 **Rigorously tested**: These individually-versioned components have been vetted to be accessible and inclusive of global audiences.
* 📱 **Multi-platform support**: We support [evergreen browsers](https://github.com/adobe/spectrum-css?tab=readme-ov-file#browser-support) for scalability and flexibility.

&nbsp;&nbsp;&nbsp;&nbsp; [<img src="https://img.shields.io/badge/Get%20started-F0F0F0?style=for-the-badge&logo=adobe&logoColor=%23FF0000"/>](https://opensource.adobe.com/spectrum-css/get-started.html) &nbsp; [<img src="https://img.shields.io/badge/Storybook-F0F0F0?style=for-the-badge&logo=storybook&logoColor=%23FF4785"/>](https://opensource.adobe.com/spectrum-css/preview/)

## Using Spectrum CSS

The preferred method of using Spectrum CSS relies on custom properties to swap out variables for different themes and contexts. This results in the lowest bundle sizes and is one of the simpler approaches.

> Want to get a component up and running ASAP? Check out the [quick start guide](https://github.com/adobe/spectrum-css/blob/main/.github/QUICK-START.md).

Reviewer: ```suggestion
Want to get a component up and running as soon as possible? Check out the [quick start guide](https://github.com/adobe/spectrum-css/blob/main/.github/QUICK-START.md).
```

Want to get a component up and running ASAP? Check out the [quick start guide](https://github.com/adobe/spectrum-css/blob/main/.github/QUICK-START.md).

### Functionality

Spectrum CSS is CSS-only, implementing only the interactivity that can be done with pure CSS. We do include some lightweight JS-based interaction for demonstration purposes only in our Storybook and documentation. Spectrum CSS should only be used by implementations of Spectrum, or very simple applications that only need things like typography, checkboxes, text fields, etc.
Spectrum CSS is CSS-only, implementing only the interactivity that can be done with pure CSS. We do include some lightweight JS-based interaction for demonstration purposes only in our [Storybook](https://opensource.adobe.com/spectrum-css/preview/). Spectrum CSS should only be used by implementations of Spectrum, or very simple applications that only need things like typography, checkboxes, text fields, etc.

> Adobe maintains separate libraries written with [web components](https://opensource.adobe.com/spectrum-web-components/) and [React](https://react-spectrum.adobe.com/react-spectrum/index.html) that work in partnership with Spectrum CSS to create fully interactive Spectrum components.

Reviewer 3: Curious why remove the link to React?

Author: It's my understanding that the React project stopped using Spectrum CSS a while back. Is that the case?

Reviewer 3: They did stop using our code but we collaborate with them. I just don't want to stop documenting them since they're an important implementation of the design system.

Author: Ok, makes sense! I'll revert this change 👍

### Installing components

Each component is released on npm as a separate, individually versioned package inside of the [@spectrum-css](https://www.npmjs.com/org/spectrum-css) org.

> To incorporate Spectrum CSS into your project, you install only the components (and their dependencies) you need:

Reviewer 3: ```suggestion
To incorporate Spectrum CSS into your project, you install the components you need along with any peer dependencies you need for your implementation. Most peer dependencies are optional and can bring additional features when installed along with a component. For example a button can be rendered with or without an icon; so if your use-case leverages an icon with your button, you'll want to install icon as well as a peer dependency.
```

```shell
yarn add -DW @spectrum-css/tokens @spectrum-css/typography @spectrum-css/page @spectrum-css/icon @spectrum-css/button
```

Installed components will be available in the `node_modules/@spectrum-css/` folder of your project.

> All components in this library have a peer dependency on [`@spectrum-css/tokens`](tokens), which is a local package that serves up the [Spectrum design tokens](https://github.com/adobe/spectrum-tokens) as CSS custom properties (via [Style dictionary](https://amzn.github.io/style-dictionary/#/)). These custom properties are leveraged in all components to create a cohesive visual language and to allow for easy theming. If you choose not to use the provided `@spectrum-css/tokens` package, you must define your own custom properties or your components will not render correctly. Overriding the tokens is not recommended, however, modifying styles is supported through the use of component-specific `--mod` prefixed properties. [More on this below](https://github.com/adobe/spectrum-css?tab=readme-ov-file#modifying-components).

Reviewer 3: ```suggestion
All components in this library have a peer dependency on [`@spectrum-css/tokens`](tokens), which is a local package that serves up the [Spectrum design tokens](https://github.com/adobe/spectrum-tokens) as CSS custom properties (via [Style dictionary](https://amzn.github.io/style-dictionary/#/)). These custom properties are leveraged in all components to create a cohesive visual language and to allow for easy theming. If you choose not to use the provided `@spectrum-css/tokens` package, you must define your own custom properties or your components will render with a significant amount of missing styles. When overriding certain styles is necessary, modifying styles is supported through the use of component-specific `--mod` prefixed properties. [More on this below](https://github.com/adobe/spectrum-css?tab=readme-ov-file#modifying-components).
```

@import "node_modules/@spectrum-css/tokens/dist/index.css";

/*

>   TODO

Reviewer 3: Is this TODO intentional? This comment block is part of the code snippet.

Author: Good catch! I'll remove

@import "node_modules/@spectrum-css/icon/dist/index.css";

/*
  Recommended: For components with multiple contexts available, if you
  want access to all contexts, load the index.css file, which includes
  all contexts and component variables.
*/
@import "node_modules/@spectrum-css/button/dist/index.css";

/*
  If you only need the spectrum visual context: For components with
  multiple contexts available, load only the spectrum context by sourcing
  index-base.css and the spectrum theme from the themes directory.
*/
@import "node_modules/@spectrum-css/button/dist/index-base.css";
@import "node_modules/@spectrum-css/button/dist/themes/spectrum.css";
```

Tokens values are mapped to context-specific classes which can be applied to the `<html>` element or any place in your DOM where you wish to encapsulate or alter the visual language of your Spectrum components.

All contexts you want to use must be defined in order to load all the appropriate custom properties for the components you are using.

#### Global variables
##### Visual language

* `.spectrum` - The default visual language for Spectrum CSS

> * `.spectrum--express` - A variant of the standard visual language. _This visual language will be deprecated in Spectrum 2_.

Reviewer 3: Nice!  Great call-out!
